src/dao/MoodDao.java

```diff
@@ -45,6 +89,34 @@ public MoodDao() {
      * executing the query
      */
     public Mood getMoodById(int moodId) throws DaoException {
```

Comment:
This method has the same problem as the previous: it doesn't return a value. The Mood is instantiated in the correct location but the return statement needs to be after the finally block.